### PR TITLE
data: rename Power category into Power Component

### PR DESCRIPTION
**Pull request by daggs1**
derived from https://github.com/endless-sky/endless-sky/pull/7906 rename Power category into Power Component so no special handling is required when that pr gets merged

motivation:
when scanning a ship, there is a need to display an entry for outfits that are unknown to the player.
a logical assumption is use the category, this means that the following will be displayed:

1. Unknown System
2. Unknown Gun
3. Unknown Torrent
4. Unknown Ammunition
5. Unknown Engine
6. Unknown Secondary Weapon
7. Unknown Hand to Hand
8. Unknown Power

the first 7 sound logical but the last one sounds like some super hero stuff...
Unknown Power Component sounds logical.

see: https://github.com/endless-sky/endless-sky/pull/7906#discussion_r1055771729

### CODE REVIEW

**Comment by Hurleveur** [comment]
You should add the old category in a depreciated folder or simply keep it with the rest and write #depreciated in a comment so we dont break plugins

**Comment by Hurleveur** [comment]
Also plz copy your explanation of why we do this from that comment on the other Pr

**Comment by daggs1** [comment]
> You should add the old category in a depreciated folder or simply keep it with the rest and write #depreciated in a comment so we dont break plugins

I didn't found any depreciated folder under data, only files, can you point me folder you are referring to?
what is the other alternative?

**Comment by daggs1** [comment]
> Also plz copy your explanation of why we do this from that comment on the other Pr

updated the original entry

File: data/categories.txt [hunk 1]
```diff
@@ -40,7 +40,7 @@ category "outfit"
 	"Secondary Weapons"
 	"Ammunition"
 	"Systems"
-	"Power"
+	"Power Component"
```
Comment (Hurleveur) [inline data/categories.txt:43]:
```suggestion
	"Power" # depreciated, should not be used anymore but kept for plugin compatibility
	"Power Component"
```
we could just do this

Comment (daggs1) [reply data/categories.txt:43]: should there be other change in the outfits file? what about the deprecated outfits?

**Review by Hurleveur — APPROVED** [review]
I think that if you had missed any we'd have had a warning from the data file check so should be good!

**Comment by daggs1** [comment]
> ~I think that if you had missed any we'd have had a warning from the data file check so should be good!~ Nvm actually we can give any category to outfits I trust you used the search function of a text editor to not miss any?

for change: sed -i 's/category "Power"/category "Power Component"/g' data/deprecated\ outfits.txt
for search: fgrep -R Power * | grep -v "Power Component"

every match in the search's output that was found after the change was inspected.
also this: ./endless-sky -o -a | grep -v ",Power Component," | grep Power returns nothing while this: ./endless-sky -o -a | grep ",Power Component," returns lots of entries

all should be ok but you can never know...

**Comment by Zitchas** [comment]
"Unknown Power" sounds like superpowers? Well, you know what they say about sufficiently advanced technology being indistinguishable from magic... And "more advanced" is generally what most unknown outfits are going to be...

As far as this goes, I don't really see a need to have this expanded out. It makes it the name longer, and doesn't really add anything to it. Keeping in mind that every one of those categories implicitly has "outfit" on the end. We don't have "power", we have "power outfits". The "outfit" word may not be spelled out, but everything in that panel is an outfit, every category is a category *of outfits.* So with this change, it will now effectively be "Power Component Outfits," which, while not bad, is a rather redundant statement.

That being said, I'd never thought of it refering to super powers, although I probably will in the future now. But that's just an amusing alternate interpretation, not really something that I feel needs to be explicitly ruled out. Is it a problem if, for a brief moment, someone thinks about superpowers when they read the outfit scanner report and see "unknown power" outfits? 

Who knows, some plug-in may add psychic powers or something that they file under this category, and now it won't fit. :)

**Comment by Hurleveur** [comment]
Just add an explicit outfit at the end of each of them in the other PR then?

**Comment by Hurleveur** [comment]
~~Unknown gun/sysem outfit sounds weird though~~
Mb just let it be superpower like

**Comment by Zitchas** [comment]
That would make more sense than having a single category that is called out as explicitly  being "components" while every other category is... not components?

As far as that goes, "components", to me, sounds like little pieces of things. Capacitors, resistors, speakers, processors, stuff that tends to be useless on its own. I know this isn't forcibly the case, but that's just what first comes to mind. Whereas everything in this category is practically a stand-alone power distribution system centered around whatever the named outfit is.

**Comment by warp-core** [comment]
Making this change for the sake of a feature that doesn't seem entirely complete to me seems inappropriate.

Additionally, is the idea that all new outfit category names must conform to this style? This is highly restrictive for no good reason.

If the category is the most basic piece of information you want to provide in the other PR, you should support defining an explicit overriding category name to be displayed where the normal name wouldn't fit your style for as yet unrecognised outfits there.

Alternatively, just provide a table:
```
Unknown outfits:
<category name>: <count>
```
Or, just leave it as is and have the first entry in each category be something like:
```
<count> Unknown
```

Regardless, not this.

**Comment by Hurleveur** [comment]
Yeah we agreed on doing something else for sure, just adding outfit at the end seems like the best solution
Just waiting on daggs1 to close this PR and do that

**Comment by Hurleveur** [comment]
Actually I'll close this since we will name all of them unknown outfits in the other PR anyway.
We could always do this later on to expand on the behavior and not just display unknown outfit but its fine honestly.

**Comment by daggs1** [comment]
this PR was a result of another pending PR.
a suggestion was raised in the other PR that makes more sense and results with a simpler solution which makes this PR redundant so closing it is the right decision.

**Pull request closed without merging**